internal/admission/defaulting_test.go

```diff
@@ -111,3 +111,142 @@ func TestTimeout(t *testing.T) {
+	}}}
+
+	type args struct {
+		labels            map[string]string
+		shouldBeValidated bool
+	}
+	type want struct {
+		patchesCount     int
+		validationStatus string
+	}
+	tests := []struct {
+		name string
+		args args
+		want want
+	}{
+		{
+			name: "pod without label should pass with validation and set Success",
+			args: args{
+				labels:            nil,
+				shouldBeValidated: true,
+			},
+			want: want{
+				patchesCount:     1,
+				validationStatus: pkg.ValidationStatusSuccess,
+			},
+		},
+		{
+			name: "pod with label Success should pass with validation",
+			args: args{
+				labels:            map[string]string{pkg.PodValidationLabel: pkg.ValidationStatusSuccess},
+				shouldBeValidated: true,
+			},
+			want: want{
+				patchesCount: 0,
+			},
+		},
+		{
+			name: "pod with label Failed should pass without validation and unchanged",
+			args: args{
+				labels:            map[string]string{pkg.PodValidationLabel: pkg.ValidationStatusFailed},
+				shouldBeValidated: false,
+			},
+			want: want{
+				patchesCount: 0,
+			},
+		},
+		{
+			name: "pod with label Reject should pass without validation and unchanged",
+			args: args{
+				labels:            map[string]string{pkg.PodValidationLabel: pkg.ValidationStatusReject},
+				shouldBeValidated: false,
+			},
+			want: want{
+				patchesCount: 0,
+			},
+		},
+		{
+			name: "pod with label Pending should pass without validation and unchanged",
+			args: args{
+				labels:            map[string]string{pkg.PodValidationLabel: pkg.ValidationStatusPending},
+				shouldBeValidated: false,
+			},
+			want: want{
+				patchesCount: 0,
+			},
+		},
+		{
+			name: "pod with unknown label should pass without validation and unchanged",
+			args: args{
+				labels:            map[string]string{pkg.PodValidationLabel: "some-unknown-label"},
+				shouldBeValidated: false,
+			},
+			want: want{
+				patchesCount: 0,
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			//GIVEN
+			imageName := "test:should-not-be-validated"
+			if tt.args.shouldBeValidated {
+				imageName = "test:should-be-validated"
+			}
+			pod := corev1.Pod{ObjectMeta: metav1.ObjectMeta{Name: "pod", Namespace: testNs},
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{{Image: imageName}}},
+			}
+			if tt.args.labels != nil {
+				pod.ObjectMeta.Labels = tt.args.labels
+			}
+
+			raw, err := json.Marshal(pod)
+			require.NoError(t, err)
+
+			req := admission.Request{
+				AdmissionRequest: admissionv1.AdmissionRequest{
+					Kind:   metav1.GroupVersionKind{Kind: PodType, Version: corev1.SchemeGroupVersion.Version},
+					Object: runtime.RawExtension{Raw: raw},
+				}}
+			client := fake.NewClientBuilder().WithScheme(scheme).WithObjects(&ns, &pod).Build()
+
+			mockValidator := mocks.ImageValidatorService{}
+			mockValidator.Mock.On("Validate", mock.Anything, "test:should-not-be-validated").
+				Panic("unexpected validation call!")
```

Comment:
Can we return error instead of panic?
The error will be catched later in `require.True(t, res.AdmissionResponse.Allowed)`

Reply:
done (in other way)